parmed/openmm/parameters.py

```diff
@@ -217,27 +212,25 @@ def write(self, dest, provenance=None, write_unused=True, separate_ljforce=False
 
         if charmm_imp:
             self._find_explicit_impropers()
-        try:
-            root = etree.Element('ForceField')
-            self._write_omm_provenance(root, provenance)
-            self._write_omm_atom_types(root, skip_types)
-            self._write_omm_residues(root, skip_residues, valid_patches_for_residue)
-            self._write_omm_patches(root, valid_residues_for_patch)
-            self._write_omm_bonds(root, skip_types)
-            self._write_omm_angles(root, skip_types)
-            self._write_omm_urey_bradley(root, skip_types)
-            self._write_omm_dihedrals(root, skip_types, improper_dihedrals_ordering)
-            self._write_omm_impropers(root, skip_types)
-#           self._write_omm_rb_torsions(root, skip_types)
-            self._write_omm_cmaps(root, skip_types)
-            self._write_omm_scripts(root, skip_types)
-            self._write_omm_nonbonded(root, skip_types, separate_ljforce)
-            self._write_omm_LennardJonesForce(root, skip_types, separate_ljforce)
-        finally:
-            string = etree.tostring(root, encoding='utf-8').decode('utf-8')
-            dest.write(string)
-            if own_handle:
-                dest.close()
```

Comment:
I think the existing I/O approach has some nice features, and I'm trying to get ParmEd to the point where it can handle either file names or file-like objects wherever it does I/O.  This lets you do stuff like leverage multiple writers to write to the same file, write directly to a tar/zip, do everything in-memory using BytesIO or StringIO, or even feed it into an open stream in the `requests` library to feed it into a web API somewhere.

Reply:
I believe [`lxml.etree.Element.write`](https://docs.python.org/2/library/xml.etree.elementtree.html#xml.etree.ElementTree.ElementTree.write) can handle both filenames and file-like objects:
> file is a file name, or a file object opened for writing.

Reply:
Ah, I didn't realize that.  `genopen` does have the added benefit that it auto-compresses files with `.gz` and `.bz2` extensions, so I like that behavior a little better than what `lxml` does (unless it does that, too!)